go/util/functions/all.go

```diff
@@ -6,16 +6,40 @@ package functions
 
 import "sync"
 
-// Runs all functions in |fs| in parallel, and returns when all functions have returned.
+// All runs all functions in parallel, and returns when all functions have
+// returned.
 func All(fs ...func()) {
 	wg := &sync.WaitGroup{}
 	wg.Add(len(fs))
-	for _, f_ := range fs {
-		f := f_
+	for _, f := range fs {
+		f := f
 		go func() {
+			defer wg.Done()
 			f()
-			wg.Done()
 		}()
 	}
 	wg.Wait()
 }
+
+// MaybeAll runs all functions in parallel, and returns when all functions have
+// returned. If any function returns an error, returns that error, but not
```

Comment:
This seems like a weird contract ... it waits until all functions have returned, but only returns one of the errors at random? It seems like you'd either want (a) it returns when the first one errors, or (b) it waits until all are done and returns all the errors.